src/proposals/mips/mip-b15/mip-b15.sol

```diff
@@ -0,0 +1,624 @@
+
+                /// Approvals
+                _pushCrossChainAction(
+                    addresses.getAddress(config.tokenAddressName),
+                    abi.encodeWithSignature(
+                        "approve(address,uint256)",
+                        cTokenAddress,
+                        config.initialMintAmount
+                    ),
+                    "Approve underlying token to be spent by market"
+                );
+
+                /// Initialize markets
+                _pushCrossChainAction(
+                    cTokenAddress,
+                    abi.encodeWithSignature(
+                        "mint(uint256)",
+                        config.initialMintAmount
+                    ),
+                    "Initialize token market to prevent exploit"
+                );
+
+                _pushCrossChainAction(
+                    cTokenAddress,
+                    abi.encodeWithSignature(
+                        "transfer(address,uint256)",
+                        address(0),
+                        1
+                    ),
+                    "Send 1 wei to address 0 to prevent a state where market has 0 mToken"
+                );
+
+                _pushCrossChainAction(
+                    unitrollerAddress,
+                    abi.encodeWithSignature(
+                        "_setCollateralFactor(address,uint256)",
+                        addresses.getAddress(config.addressesString),
+                        config.collateralFactor
+                    ),
+                    "Set Collateral Factor for MToken market in comptroller"
+                );
+            }
+        }
+
+        /// -------------- EMISSION CONFIGURATION --------------
+
+        EmissionConfig[] memory emissionConfig = getEmissionConfigurations(
+            block.chainid
+        );
+        MultiRewardDistributor mrd = MultiRewardDistributor(
+            addresses.getAddress("MRD_PROXY")
+        );
+
+        unchecked {
+            for (uint256 i = 0; i < emissionConfig.length; i++) {
+                EmissionConfig memory config = emissionConfig[i];
+
+                _pushCrossChainAction(
+                    address(mrd),
+                    abi.encodeWithSignature(
+                        "_addEmissionConfig(address,address,address,uint256,uint256,uint256)",
+                        MToken(addresses.getAddress(config.mToken)),
+                        addresses.getAddress(config.owner),
+                        config.emissionToken,
+                        config.supplyEmissionPerSec,
+                        config.borrowEmissionsPerSec,
+                        config.endTime
+                    ),
+                    "Add emission config for MToken market in MultiRewardDistributor"
+                );
+            }
+        }
+    }
+
+    function run(
+        Addresses addresses,
+        address
+    ) public override(CrossChainProposal, MIPProposal) {
+        printCalldata(addresses);
```

Comment:
I don't think we need to call printCalldata here, just call this proposal with DO_PRINT=true